packages/model-viewer/src/three-components/ModelUtils.ts

```diff
@@ -36,10 +32,35 @@ export const cloneGltf = (gltf: GLTF): GLTF => {
    */
   const cloneAndPatchMaterial = (material: Material): Material => {
     const clone = material.clone();
+    // This allows us to patch three's materials, on top of patches already
+    // made, for instance GLTFLoader patches SpecularGlossiness materials.
+    // Unfortunately, three's program cache differentiates SpecGloss materials
+    // via onBeforeCompile.toString(), so these two functions do the same thing
+    // but look different in order to force a proper recompile.
+    const oldOnBeforeCompile = material.onBeforeCompile;
+    clone.onBeforeCompile = (material as any).isGLTFSpecularGlossinessMaterial ?
+        (shader: Shader) => {
+          oldOnBeforeCompile(shader, undefined as any);
+          shader.fragmentShader = shader.fragmentShader.replace(
+              '#include <alphatest_fragment>', alphaChunk);
+        } :
+        (shader: Shader) => {
+          shader.fragmentShader = shader.fragmentShader.replace(
+              '#include <alphatest_fragment>', alphaChunk);
+          oldOnBeforeCompile(shader, undefined as any);
+        };
+    // This makes shadows better for non-manifold meshes
     clone.shadowSide = FrontSide;
+    // This improves transparent rendering and can be removed whenever
+    // https://github.com/mrdoob/three.js/pull/18235 finally lands.
     if (clone.transparent) {
       clone.depthWrite = false;
     }
+    // This little hack ignores alpha for opaque materials, in order to comply
+    // with the glTF spec.
+    if (!clone.alphaTest && !clone.transparent) {
+      clone.alphaTest = -0.5;
```

Comment:
Does `-0.5` have meaningful significance?

Reply:
No, any value less than zero has the same behavior.